src/Hive.IO/Hive.IO/Building/Components.cs

```diff
@@ -188,6 +188,19 @@ public Wall(rg.BrepFace surfaceGeometry) : base(surfaceGeometry)
         }
         [JsonConstructor] 
         protected Wall() { }
+
+        public override void ApplySia2024Construction(Sia2024Record siaRoom)
+        {
+
+            OpaqueConstruction opaqueConstruction = new OpaqueConstruction("SIA2024_Opaque")
+            {
+                UValue = siaRoom.UValueWalls
+            };
+            this.SpecificCo2 = siaRoom.EmissionsWalls;
+            this.SpecificCost = siaRoom.CostWalls;
+            this.Construction = opaqueConstruction;
+        }
+
```

Comment:
Wouldn't it make sense to move this function to the base class `Component` and add an argument which specifies which siaRoom data it should read? There's some code repetition going on otherwise, whereas it really is doing the same for all component (Roof, Wall, ...)

Reply:
just saw, the method is already in the base class.
and on a second thought... then you'd have to provide that argument everywhere... let's leave it as it is